askama_shared/src/generator.rs

```diff
@@ -1688,7 +1701,12 @@ impl<'a, S: std::hash::BuildHasher> Generator<'a, S> {
     // prefix whitespace suppressor from the given argument, flush that whitespace.
     // In either case, `next_ws` is reset to `None` (no trailing whitespace).
     fn flush_ws(&mut self, ws: Ws) {
-        if self.next_ws.is_some() && !ws.0 {
+        if self.next_ws.is_none() {
+            return;
+        }
+        // If `automatic_block_trim` is enabled, we keep the whitespaces only if there is a `-`.
```

Comment:
This comment, too. While you're in there, mind adding an empty line after the `}` above?